packages/typescript-checker/src/typescript-compiler.ts

```diff
@@ -0,0 +1,268 @@
+    const compiler = ts.createSolutionBuilderWithWatch(host, [this.tsconfigFile], {});
+    compiler.build();
+    return await this.check([]);
+  }
+
+  public async check(mutants: Mutant[]): Promise<ts.Diagnostic[]> {
+    this.lastMutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file!.resetMutant();
+    });
+    mutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file!.mutate(mutant);
+    });
+    await this.currentTask.promise;
+    const errors = this.currentErrors;
+    this.currentTask = new Task();
+    this.currentErrors = [];
+    this.lastMutants = mutants;
+    return errors;
+  }
+
+  public get nodes(): Map<string, TSFileNode> {
+    if (!this._nodes.size) {
+      // create nodes
+      for (const [fileName] of this.sourceFiles) {
+        const node = new TSFileNode(fileName, [], []);
+        this._nodes.set(fileName, node);
+      }
+
+      // set children
+      for (const [fileName, file] of this.sourceFiles) {
+        const node = this._nodes.get(fileName);
+        if (node == null) {
+          throw new Error(
+            `Node for file '${fileName}' could not be found. This should not happen. This shouldn't happen, please open an issue on the stryker-js github`
+          );
+        }
+
+        const importFileNames = [...file.imports];
+        node.children = importFileNames.map((importName) => this._nodes.get(importName)!).filter((n) => n != undefined);
+      }
+
+      // set parents
+      for (const [, node] of this._nodes) {
+        node.parents = [];
+        for (const [, n] of this._nodes) {
+          if (n.children.includes(node)) {
+            node.parents.push(n);
+          }
+        }
+      }
+    }
+
+    return this._nodes;
+  }
+
+  /**
+   * Resolves TS input file based on a dependency of a input file
+   * @param dependencyFileName The dependency file name. With TS project references this can be a declaration file
+   * @returns TS source file if found (fallbacks to input filename)
+   */
+  private resolveTSInputFile(dependencyFileName: string): string {
+    if (!dependencyFileName.endsWith('.d.ts')) {
+      return dependencyFileName;
+    }
+
+    const file = this.fs.getFile(dependencyFileName);
+    if (!file) {
+      throw new Error(`Could not find ${dependencyFileName}`);
+    }
+
+    const sourceMappingURL = getSourceMappingURL(file.content);
+    if (!sourceMappingURL) {
+      return dependencyFileName;
+    }
+
+    const sourceMapFileName = path.resolve(dependencyFileName, '..', sourceMappingURL);
+    const sourceMap = this.fs.getFile(sourceMapFileName);
+    if (!sourceMap) throw new Error(`Could not find ${sourceMapFileName}`);
+
+    const sources: string[] = JSON.parse(sourceMap.content).sources;
+
+    if (sources.length === 1) {
+      const sourcePath = sources[0];
+      return toPosixFileName(path.resolve(sourceMapFileName, '..', sourcePath));
+    }
+
+    return dependencyFileName;
```

Comment:
This part has no code coverage. We should add a TS solution with `.tsbuildinfo` files and project references to the `testResources` directory and use that to test some of these code paths.